### ISSUE TITLE
[@typescript-eslint/consistent-type-definitions] interface cannot be used as a type parameter in a special case

### ISSUE THREAD

**Issue by Jamesgt** [typescript-eslint/typescript-eslint#2776]
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/<rule>": ["<setting>"]
  }
}
```

```TS
// test @typescript-eslint/consistent-type-definitions

type T = {
  k: number;
}

interface I1 {
  k: number;
}

interface I2 extends Record<string, unknown> {
  k: number;
}

type W<P extends Record<string, unknown>> = {
  x: P;
}

type TestType = W<T>;
type TestInterface1 = W<I1>; // rule does not allow this
type TestInterface2 = W<I2>;

const testType: TestType = { x: { k: 1, extraKey: 3 } }; // extraKey is not allowed, as intended
const testInterface1: TestInterface1 = { x: { k: 1, extraKey: 3 } };
const testInterface2: TestInterface2 = { x: { k: 1, extraKey: 3 } }; // extraKey is allowed, because I asked it to be, but this is not the original intent

class CT implements T { k: number = 0; }

class CI implements I1 { k: number = 0; }
```

tsconfig:
```
"@typescript-eslint/consistent-type-definitions": [
      "error",
      "interface"
    ],
```

**Expected Result**

if `type` and `interface` are just "stylistic" decisions then this `type TestInterface1 = W<I1>;` should compile

**Actual Result**

type TestInterface1 = W<I1>; // rule does not allow this

**Additional Info**

-

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.7.0` |
| `@typescript-eslint/parser`        | `4.7.0` |
| `TypeScript`                       | `4.0.5` |
| `ESLint`                           | `7.13.0` |
| `node`                             | `14.7.0` |

**Comment by bradzacher** [comment]
Why can't you just do the following?

```ts
interface TestInterface1 extends W {}
```

The rule is setup to enforce one way or the other.
If you have a "special case" where you want to use a type declaration - then you must use a disable comment.
